lib/rugged_adapter/git_layer_rugged.rb

```diff
@@ -188,13 +188,29 @@ def cat_file(options, sha)
         @repo.lookup(sha).read_raw
       end
 
-      def apply_patch(head_sha = 'HEAD', patch=nil)
-        false # Rewrite gollum-lib's revert so that it doesn't require a direct equivalent of Grit's apply_patch
-      end
-
-      def revert(path, sha1, sha2, ref)
-        # FIXME: See https://github.com/gollum/grit_adapter/pull/14
-        fail NotImplementedError
+      def revert(path, sha1, sha2)
+        options = {}
+        options[:paths] = [path] if path
+        patch = path ?
+          @repo.diff(sha2, sha1, options).first.diff : @repo.diff(sha2, sha1)
+        options[:location] = :index
+        files = []
+        if path
+          files << path
+        else
+          patch.each_delta do |delta|
+            files << delta.new_file[:path]
+            files << delta.old_file[:path]
+          end
+        end
+        files.uniq!
+        begin
+          result = @repo.apply(patch, options)
+        rescue RuntimeError
+          result = false
+        end
+        return false unless result
```

Comment:
Just for clarification: do I presume correctly that this line is here because `@repo.apply(patch, options)` can return `nil` ?

Reply:
`@repo.apply` will return either `true`, or throw an exception. So I now see that we can get rid of `result` and just `return false` when the exception is caught. Thanks for noticing this is weird!